homeassistant/components/light/greenwave.py

```diff
@@ -0,0 +1,108 @@
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/light.tcpbulbs/
+"""
+import logging
+import voluptuous as vol
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, Light, PLATFORM_SCHEMA, SUPPORT_BRIGHTNESS)
+from homeassistant.const import CONF_HOST
+import homeassistant.helpers.config_validation as cv
+
+SUPPORTED_FEATURES = (SUPPORT_BRIGHTNESS)
+REQUIREMENTS = ['greenwavereality==0.2.9']
+_LOGGER = logging.getLogger(__name__)
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required("version"): cv.positive_int,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup TCP Connected Platform."""
+    import greenwavereality as greenwave
+    import os
+    host = config.get(CONF_HOST)
+    tokenfile = hass.config.path('greenwave.token')
+    if config.get("version") == 3:
+        if os.path.exists(tokenfile):
+            tokenfile = open(tokenfile)
+            token = tokenfile.read()
+            tokenfile.close()
+        else:
+            token = greenwave.grab_token(host, 'hass', 'homeassistant')
+            tokenfile = open(tokenfile, "w+")
+            tokenfile.write(token)
+            tokenfile.close()
+    else:
+        token = None
+    doc = greenwave.grab_xml(host, token)
+    add_devices(TcpLights(device, host, token) for device in doc)
+
+
+class TcpLights(Light):
+    """Representation of an TCP Connected Light."""
+
+    def __init__(self, light, host, token):
+        """Initialize a TCP Connected Light."""
+        import greenwavereality as greenwave
+        self._did = light['did']
+        self._name = light['name']
+        self._state = int(light['state'])
+        self._brightness = greenwave.hass_brightness(light)
+        self._host = host
+        self._online = greenwave.check_online(light)
+        self.token = token
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return SUPPORTED_FEATURES
+
+    @property
+    def available(self):
+        """Return True if entity is available."""
+        return self._online
+
+    @property
+    def name(self):
+        """Return the display name of this light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def is_on(self):
+        """Return true if light is on."""
+        return self._state
+
+    def turn_on(self, **kwargs):
+        """Instruct the light to turn on."""
+        import greenwavereality as greenwave
+        temp_brightness = int((kwargs.get(ATTR_BRIGHTNESS, 255)
+                               / 255) * 100)
+        greenwave.set_brightness(self._host, self._did,
+                                 temp_brightness, self.token)
+        greenwave.turn_on(self._host, self._did, self.token)
```

Comment:
You are sure that is also need to send a brightness and after that also a on?

Reply:
Yes, both are required.